### ISSUE TITLE
Play continues after a failed task when `any_errors_fatal` is set

### ISSUE THREAD

**Issue by chlunde** [ansible/ansible#13744]
##### Issue Type

Bug Report
##### Ansible Version

2.0 from the "stable-2.0" branch, commit fcd074d40f32ef52ed286678293984125ff51008

```
$ ansible --version
ansible 2.0.0
  config file =
  configured module search path = Default w/o overrides
```
##### Ansible Configuration

No changes
##### Environment

RHEL 7
##### Summary

Play continues after a failed task when `any_errors_fatal` is set.
##### Steps To Reproduce:

```
$ cat inv
app01
app02

$ ansible-playbook -i inv stop.yml
```

``` yaml

---
- hosts: all
  any_errors_fatal: true
  max_fail_percentage: 0
  serial: 100%
  tasks:
    - name: "shell command failing on one node"
      shell: "test $(hostname --short) == app01"

    - name: stop before this
      command: echo noooooooo
```
##### Expected Results:

I expected the playbook to stop after the executing the first task on all nodes, before the "stop before this" task, as it does on 1.9.4.  The documentation says:

> The `any_errors_fatal` play option will mark all hosts as failed if any fails, causing an immediate abort"
##### Actual Results:

The second task is executed on the node which completed the first task without error:

```
$ ansible-playbook -i inv stop.yml

PLAY ***************************************************************************

TASK [setup] *******************************************************************
ok: [app02]
ok: [app01]

TASK [shell command failing on one node] ***************************************
fatal: [app02]: FAILED! => {"changed": true, "cmd": "test $(hostname --short) == app01", "delta": "0:00:00.005278", "end": "2016-01-07 01:38:57.751466", "failed": true, "rc": 1, "start": "2016-01-07 01:38:57.746188", "stderr": "", "stdout": "", "stdout_lines": [], "warnings": []}
changed: [app01]

TASK [stop before this] ********************************************************
changed: [app01]

PLAY RECAP *********************************************************************
app01                      : ok=3    changed=2    unreachable=0    failed=0
app02                      : ok=1    changed=0    unreachable=0    failed=1
```

**Comment by nocive** [comment]
seeing this behaviour as well when using `fail` module in a playbook with `any_errors_fatal: true`

**Comment by alangibson** [comment]
This also occurs on CentOS Linux release 7.0.1406 (Core) with ansible 2.0.0.1 (detached HEAD a50d1ea756) run from source. It seems like the flag is simply ignored in v2 because this was working on v1.9.

**Comment by jimi-c** [comment]
# Closing This Ticket

Hi!

We believe the above commit should resolve this problem for you. This will also be included in the next 2.0 release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!

**Comment by stivio** [comment]
currently doesn't work at stable-2.2 and devel, working with stable-2.1

**Comment by marquetemb** [comment]
Doesn't work at stable-2.2

**Comment by elouanKeryell-Even** [comment]
Still doesn't seem to work with ansible 2.3

**Comment by alangibson** [comment]
It looks like this might be fixed in 2.4.0: https://github.com/ansible/ansible/issues/22924

**Comment by csillab** [comment]
I am running 2.3.1.0 and it still seem to be the case.

**Comment by ktosiek** [comment]
Core team mostly ignores closed issues, if anyone is still seeing this problem please open a new one (and leave a message about it here, for those who are still interested but have less time :-))

**Comment by ml0renz0** [comment]
Still happening...

**Comment by hmajid2301** [comment]
Upgrading to ansible 2.4.0 (`sudo pip install git+git://github.com/ansible/ansible.git@devel`) solved this issue for me.

**Comment by abhijeetka** [comment]
The fix is present in ansible 2.4.1.
